### ISSUE TITLE
ECS Service creation hangs

### ISSUE THREAD

**Issue by ajcrites** [hashicorp/terraform#5454]
My `.tf` file looks something like:

```
provider "aws" {
  region = "us-east-1"
}

resource "aws_ecr_repository" "repo" {
  name = "repo"
  provisioner "local-exec" {
    command = "provision/deploy-container.sh"
  }
}

resource "aws_ecs_task_definition" "task" {
  family = "task"
  container_definitions = <<EOF
[
  {
    "name": "ingest",
    "image": "https://${aws_ecr_repository.ingest-repo.registry_id}.dkr.ecr.us-east-1.amazonaws.com/ingest:latest",
    "cpu": 10,
    "memory": 500,
    "essential": true
  }
]
EOF

}

resource "aws_ecs_service" "ingest-service" {
  name = "ingest-service"
  task_definition = "${aws_ecs_task_definition.ingest-task.arn}"
  desired_count = 1
}
```

Everything up to the ECS service works -- the repo and the task definition are created appropriately. At that point, I get the following output:

```
aws_ecs_service.ingest-service: Creating...
  cluster:                            "" => "<computed>"
  deployment_maximum_percent:         "" => "200"
  deployment_minimum_healthy_percent: "" => "100"
  desired_count:                      "" => "1"
  name:                               "" => "ingest-service"
  task_definition:                    "" => "arn:aws:ecs:us-east-1:<ID>:task-definition/ingest-task:4"
```

This looks good, but at this point terraform just hangs. The ECS is not created.

If I press Ctrl-c I get

> Interrupt received. Gracefully shutting down...

but it still hangs for quite a long time ... seemingly indefinitely.

**Comment by phinze** [comment]
Thanks for the report, @ajcrites.

Can you capture a [debug log](https://www.terraform.io/docs/internals/debugging.html) and toss it into a [gist](https://gist.github.com)? That should help us narrow down what could be happening here.

**Comment by Bjorn248** [comment]
on behalf of @ajcrites [https://gist.github.com/Bjorn248/f197f2f5aa103752e79f](https://gist.github.com/Bjorn248/f197f2f5aa103752e79f)

**Comment by stack72** [comment]
I was able to confirm (using a recreation of your config) that this has been fixed by #5460 

The config I used was as follows:

```
provider "aws" {
  region = "us-east-1"
}

resource "aws_ecr_repository" "ingest-repo" {
  name = "repo"
  provisioner "local-exec" {
    command = "provision/deploy-container.sh"
  }
}

resource "aws_ecs_cluster" "default" {
    name = "terraformecstestcluster"
}

resource "aws_ecs_task_definition" "ingest-task" {
  family = "task"
  container_definitions = <<EOF
[
  {
    "name": "ingest",
    "image": "https://${element(split(":", aws_ecr_repository.ingest-repo.arn), 4)}.dkr.ecr.us-east-1.amazonaws.com/ingest:latest",
    "cpu": 10,
    "memory": 500,
    "essential": true
  }
]
EOF

}

resource "aws_ecs_service" "ingest-service" {
  name = "ingest-service"
  task_definition = "${aws_ecs_task_definition.ingest-task.arn}"
  desired_count = 1
  cluster = "${aws_ecs_cluster.default.name}"
}
```

**Comment by ajcrites** [comment]
Confirmed fixed in new release.
